packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/dynamic/DynamicMutableRealmObjectTests.kt

```diff
@@ -1354,6 +1386,191 @@ class DynamicMutableRealmObjectTests {
         }
     }
 
+    @Test
+    fun testSetsInRealmAny() {
+        val dynamicSampleInner = dynamicMutableRealm.copyToRealm(
+            DynamicMutableRealmObject.create(
+                "Sample",
+                "stringField" to "INNER"
+            )
+        )
+        dynamicMutableRealm.copyToRealm(
+            DynamicMutableRealmObject.create(
+                "Sample",
+                "nullableRealmAnyField" to RealmAny.create(
+                    realmSetOf(
+                        RealmAny.create(
+                            dynamicSampleInner
+                        )
+                    )
+                )
+            )
+        ).let {
+            val set = it.getNullableValue<RealmAny>("nullableRealmAnyField")!!.asSet()
+            // Verify that we get mutable instances out of the set
+            val setObject = set.first()!!.asRealmObject<DynamicMutableRealmObject>()
+            assertIs<DynamicMutableRealmObject>(setObject)
```

Comment:
Is there a test anywhere where we make sure to only allow `DynamicRealmObject` in RealmAny's associated with Dynamic Realms? (I might have missed it)

Reply:
There are a couple of test around:
- https://github.com/realm/realm-kotlin/blob/main/packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/dynamic/DynamicRealmObjectTests.kt#L1408
- https://github.com/realm/realm-kotlin/blob/main/packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/dynamic/DynamicRealmObjectTests.kt#L1507
- https://github.com/realm/realm-kotlin/blob/main/packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmAnyTests.kt#L1637

It should be the exact same code as the collection-operators are the same, but I added explicit asserts in my test too for completeness.